### PR TITLE
add exclude modules argument to map

### PR DESCRIPTION
**Pull request by Vaishaal**
Exclude modules in map based on prefix

Fix to #136

### CODE REVIEW

**Comment by ooq** [comment]
Let's maybe have an example to document this?

**Comment by Vaishaal** [comment]
Here is an example


library.py:
```
import big_module
def foo(x):
      return big_module.foo(x)

def bar(x):
       return x
```

driver.py:
```
import library 
pwex = pywren.default_executor()
pwex.map(library.bar, [0,1,2,3])
```

**Comment by Vaishaal** [comment]
I've also added a test. @ericmjonas  could you see if this is sufficient?

File: tests/test_modules.py [hunk 1]
```diff
@@ -126,4 +126,27 @@ def test_simple(self):
...
+            return extmodule.foo_add(x)
+        x = 1.0
+        fut0 = self.wrenexec.map(foo, [x])
+        res = fut0[0].result()
+        self.assertEqual(res, 2.0)
+
+        fut = self.wrenexec.map(foo, [x], exclude_modules=["extmodule"])
+        try:
+            fut[0].result()
+            self.fail("shouldn't happen")
```
Comment (ooq) [inline tests/test_modules.py]: If exclusion does not work, there might still be some exception because extmodule does not really exist right?

Comment (Vaishaal) [inline tests/test_modules.py]: What do you mean? In this case extmodule exists right?

Comment (ooq) [inline tests/test_modules.py]: Did you import it?

Comment (Vaishaal) [inline tests/test_modules.py]: at the top. line 6.

Comment (ooq) [inline tests/test_modules.py]: Ah okay, I missed it.

**Comment by shivaram** [comment]
I think this looks fine to me. @ooq do you want to take one final look ?

**Comment by ooq** [comment]
Looks good. Will merge.

**Pull request merged**